packages/pigment-css-react/src/Hidden.jsx

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable react/jsx-filename-extension */
+import * as React from 'react';
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import { generateAtomics } from './generateAtomics';
+
+const hiddenAtomics = generateAtomics(({ theme }) => {
+  const conditions = {};
+
+  for (let i = 0; i < theme.breakpoints.keys.length; i += 1) {
+    const breakpoint = theme.breakpoints.keys[i];
+    conditions[`${theme.breakpoints.keys[i]}Only`] = theme.breakpoints.only(breakpoint);
+    conditions[`${theme.breakpoints.keys[i]}Up`] = theme.breakpoints.up(breakpoint);
+    conditions[`${theme.breakpoints.keys[i]}Down`] = theme.breakpoints.down(breakpoint);
+  }
+
+  return {
+    conditions,
+    properties: {
+      display: ['none'],
+    },
+  };
+});
+
+const Hidden = React.forwardRef(function Hidden(
+  { className, component = 'div', style, ...props },
+  ref,
+) {
+  const rest = {};
+  const breakpointProps = {};
+  Object.keys(props).forEach((key) => {
+    if (key.endsWith('Up') || key.endsWith('Down')) {
+      breakpointProps[key] = 'none';
+    } else if (key === 'only') {
+      if (typeof props[key] === 'string') {
+        breakpointProps[`${props[key]}Only`] = 'none';
+      }
+      if (Array.isArray(props[key])) {
+        props[key].forEach((val) => {
+          breakpointProps[`${val}Only`] = 'none';
+        });
+      }
+    } else {
+      rest[key] = props[key];
+    }
+  });
+  const stackClasses = hiddenAtomics({ display: breakpointProps });
+  const Component = component;
+  return (
+    <Component
+      ref={ref}
+      className={clsx(stackClasses.className, className)}
+      style={{ ...style, ...stackClasses.style }}
+      {...rest}
+    />
+  );
+});
+
+Hidden.propTypes = {
```

Comment:
You'll also need to wrap this with `process.env.NODE_ENV` check

Reply:
I know that this is an edge case, but what will happen if multiple props are specified, like

`<Hidden smUp xlDown />`

Reply:
Good point, this works as expected.

Reply:
See `apps/pigment-css-next-app/src/app/hidden/page.tsx`

Reply:
> You'll also need to wrap this with process.env.NODE_ENV check

Done.